js/msp/MSPHelper.js

```diff
@@ -957,7 +958,7 @@ MspHelper.prototype.process_data = function(dataHandler) {
             dataHandler.callbacks.splice(i, 1);
 
             // fire callback
-            if (callback) callback({'command': code, 'data': data, 'length': data.byteLength});
+            if (callback) callback({'command': code, 'data': data, 'length': data.byteLength, 'crcError': crcError});
```

Comment:
The callback from `MspHelper.prototype.dataflashRead` in js/msp/MSPHelper.  But now that it's in the response object all the MSP message callbacks could use it for future implementations.  There might be other message types that could also benefit from knowing about a CRC error and retry.

But now that I'm thinking about it, it might make sense to put some logic in js/msp.js to only set the flag for specific message types that expect to deal with crc errors.  Because before the message payload was dropped (nulled) if a crc error occurred.  The callback was still called and the handler couldn't tell that there was actually a crc error and this caused the javascript runtime errors since the dataflash callback wasn't expecting the payload object to be null.  The difference is now that the payload is still returned even though it fails the crc check.  This might cause problems to other msp message type handlers in the (unlikely) event they get a crc error.

Maybe I should revise this slightly to still null out the payload (like the old crc handling), keep the flag to pass to the callbacks, and then refactor the `MspHelper.prototype.dataflashRead` callback to check the `crcError` flag before accessing the payload.  Thoughts?